### ISSUE TITLE
关于_format_bbox函数中并未将bbox中心点移回中心位置的困惑

### ISSUE THREAD

**Issue by kingofstu** [HuangJunJie2017/BEVDet#371]
作者您好，在bevdet-dev2.1中，self.pts_bbox_head.get_bboxes函数里，即centerpoint_head.get_bboxes中，将bboxes的中心移到了底面中心，可以理解这是为了送入LiDARInstance3DBoxes类：
for i in range(num_samples):
    for k in rets[0][i].keys():
        if k == 'bboxes':
            bboxes = torch.cat([ret[i][k] for ret in rets])
            bboxes[:, 2] = bboxes[:, 2] - bboxes[:, 5] * 0.5        # 移至底面中心
            bboxes = img_metas[i]['box_type_3d'](
                bboxes, self.bbox_coder.code_size)
但是在dataset.evaluate时，_format_bbox函数中，并没有将bboxes中心点移回重心的位置就送入了NuScenesBox类，这有合理的解释吗？我看’1.0.0rc4‘的mmdetection3d官方的nuscenes_dataset.py中的_format_bbox函数是要移回重心位置的，所以感到疑惑。

**Comment by jbwang1997** [comment]
我的理解是在 CenterHead 中模型直接学习 lidar_boxes的 gravity center 。[代码见此处。]( https://github.com/HuangJunJie2017/BEVDet/blob/dev3.0/mmdet3d/models/dense_heads/centerpoint_head.py#L456) 不确定是否正确，欢迎讨论。

**Comment by kingofstu** [comment]
> 我的理解是在 CenterHead 中模型直接学习 lidar_boxes的 gravity center 。[代码见此处。](https://github.com/HuangJunJie2017/BEVDet/blob/dev3.0/mmdet3d/models/dense_heads/centerpoint_head.py#L456) 不确定是否正确，欢迎讨论。

您好，是的，在训练作损失时确实是使用gravity center；但在作评估之前，会调用centerhead的get _bboxes函数，这里为了送入LiDARInstance3DBoxes类，会将box中心移至box底面中心。
[代码见此处。](https://github.com/HuangJunJie2017/BEVDet/blob/f71858d02eb0fbd09860150ade67558d7984b1be/mmdet3d/models/dense_heads/centerpoint_head.py#L751)欢迎继续讨论。

**Comment by jbwang1997** [comment]
这确实还挺奇怪的,mmdet3d 模型后处理完的 box 的 center 好像是底面中心点.我同时看了一下 [nuscenes 的 metric 说明](https://www.nuscenes.org/object-detection/?externalData=all&mapData=all&modalities=Any), 发现所有的 metric 都是在 2D 平面上去完成的.z 轴上的差异好像对最终的 metric 没影响.

**Comment by CesarLiu** [comment]
https://github.com/HuangJunJie2017/BEVDet/blob/26144be7c11c2972a8930d6ddd6471b8ea900d13/mmdet3d/datasets/nuscenes_dataset.py#L362, 原本mmdet3d的NuscenesDataset是用的gravity_center吧？也对应nuscenes-devkit的定义。虽然evaluate没啥问题，但是viz应该会有问题？毕竟是bottom_center
